homeassistant/components/device_tracker/asuswrt.py

```diff
@@ -117,272 +61,41 @@ def __init__(self, config):
         self.mode = config[CONF_MODE]
         self.port = config[CONF_PORT]
         self.require_ip = config[CONF_REQUIRE_IP]
+        self.last_results = {}
+        self.success_init = False
+        self.connection = None
 
-        if self.protocol == 'ssh':
-            self.connection = SshConnection(
-                self.host, self.port, self.username, self.password,
-                self.ssh_key)
-        else:
-            self.connection = TelnetConnection(
-                self.host, self.port, self.username, self.password)
+    async def async_connect(self):
+        """Setup connection."""
+        from aioasuswrt.asuswrt import AsusWrt
+
+        self.connection = AsusWrt(self.host, self.port,
+                                  False if self.protocol == 'ssh' else True,
+                                  self.username, self.password, self.ssh_key,
+                                  self.mode, self.require_ip)
 
         self.last_results = {}
```

Comment:
Do we need to reset this here? It's already initialized in init.

Reply:
This was left.

Reply:
🙈

Reply:
oops

Reply:
https://github.com/home-assistant/home-assistant/pull/17724